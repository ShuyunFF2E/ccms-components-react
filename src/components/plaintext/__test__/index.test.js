import React, { Component } from 'react';
import { render, mount } from 'enzyme';
// eslint-disable-next-line
import mountTest from '@tests/shared/mountTest';
import Plaintext from '../index';
import Icon from '../../icon';

describe('Plaintext', () => {
  mountTest(Plaintext);
  it('render correctly', () => {
    const wrapper = render(<Plaintext text="李子最美" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be plain', () => {
    const wrapper = render(<Plaintext text="李子最美呀" isPlain />);
    expect(wrapper).toMatchSnapshot();
  });

  it('viewPlainClick event correctly', () => {
    class PlaintextComponent extends Component {
      state = {
        content: '李**',
      };

      handleViewPlainClick = () => {
        this.setState({
          content: '李子最美',
        });
      };

      render() {
        return (
          <Plaintext
            text={this.state.content}
            onViewPlainClick={this.handleViewPlainClick}
          />
        );
      }
    }
    const wrapper = mount(<PlaintextComponent />);
    wrapper.find('Icon').simulate('click');
    expect(wrapper.state().content).toBe('李子最美');
  });

  it('ViewSecretClick event correctly', () => {
    class PlaintextComponent extends Component {
      state = {
        content: '李子**',
      };

      handleViewSecretClick = () => {
        this.setState({
          content: '李子不美',
        });
      };

      render() {
        return (
          <Plaintext
            text={this.state.content}
            isPlain
            onViewSecretClick={this.handleViewSecretClick}
          />
        );
      }
    }
    const wrapper = mount(<PlaintextComponent />);
    wrapper.find(Icon).simulate('click');
    expect(wrapper.state().content).toBe('李子不美');
  });
});
