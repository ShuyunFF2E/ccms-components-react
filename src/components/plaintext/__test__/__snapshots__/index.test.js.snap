// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plaintext it should be plain 1`] = `
<div
  class="cloud-plaintext"
>
  <i
    class="cloud-icon cloud-icon-hide"
  />
  <span>
    李子最美呀
  </span>
</div>
`;

exports[`Plaintext render correctly 1`] = `
<div
  class="cloud-plaintext"
>
  <i
    class="cloud-icon cloud-icon-view"
  />
  <span>
    李子最美
  </span>
</div>
`;
